llms/mlx_lm/models/base.py

```diff
@@ -60,3 +55,26 @@ def from_dict(cls, params):
                 if k in inspect.signature(cls).parameters
             }
         )
+
+
+def create_additive_causal_mask(N: int, offset: int = 0):
+    if offset > 0:
```

Comment:
Here I'm keeping the existing code for creating an offset mask in order to minimize change. An alternative would be to slice the output of `nn.MultiHeadAttention.create_additive_causal_mask`:

```
def create_additive_causal_mask(N: int, offset: int = 0):
    return nn.MultiHeadAttention.create_additive_causal_mask(N)[offset:,]
```